dask/dataframe/core.py

```diff
@@ -1047,6 +1047,13 @@ def apply_concat_apply(args, chunk=None, aggregate=None, columns=None):
     """
     if not isinstance(args, (tuple, list)):
         args = [args]
+
+    def concat(args):
+        if isinstance(args[0], pd.Index):
+            concat = lambda a: reduce(lambda b, c: b.append(c), a)
+        else:
+            concat = pd.concat
+        return concat(args)
```

Comment:
@cowlicks that's much better than my suggestion to `reduce`, which would only work if you didn't care whether there were duplicates (`union` vs. `union all` in SQL).